### DISCUSSION TITLE
關於在文章自動播放gif

### DISCUSSION THREAD

**Discussion by Peckystudios** [CaiJimmy/hugo-theme-stack#909] [Q&A]
大家好
不好意思再問一個問題

因為我是做動畫的，會需要直接展示動圖
目前會遇到2個問題
1. 如果有轉換成功，gif需要點開來才會播放
2. 目前因為有更新主題所以是整個網站是壞的，然後我發現他把gif變成一張圖檔，也沒辦法點開

範例可參考這裡
https://peckystudios.com/p/aetips-qq/
的這張圖
<img width="440" alt="image" src="https://github.com/CaiJimmy/hugo-theme-stack/assets/84195047/d7fde5e2-b784-4473-ae58-4accb4b3ad8d">

我自己希望可以呈現的效果：
閱讀文章時直接會看到正在動的gif（無限播放）

請問可以怎麼做呢？

**Comment by CaiJimmy** [comment]
这个问题应该是因为主题生成了 GIF 的缩略图，而 Hugo 自带的图片缩放功能会导致 GIF 变成静态图片。 

可以试试看切换到 `canary` 版本的主题，我前段时间提交了代码用来跳过 GIF 图片处理：https://github.com/CaiJimmy/hugo-theme-stack/commit/8dc2880a9c7d2e026cda92eb2cd780acfd0772c4

**Reply by Peckystudios** [reply]
謝謝你回覆！
不好意思因為我不是工程師所以不太確定怎麼更改主題版本

目前我的理解是我使用主題的方式是git的sub module （雖然我搞不清楚實際上的運作邏輯）
我記得我操作的時候就是使用git的更新指令，好像也沒有什麼主題版本的選項可以選

canary我有努力找了一下，目前理解是他是其中一個merge的版本？可能理解有大錯誤請見諒😭

另外如果我主題更改之後，是直接在config檔案加上白名單的數值嗎？就是#885上面寫的，後面加上官方提供的格式副檔名，不確定要不要加括號［］
這個有沒有特別要放在哪個項目下面？
例如menu、sidebar之類的

有空再回覆就好，真的謝謝你🙏
